File: MavericksBank/MavericksBank/Models/Transaction.cs
using System;
using System.Collections.Generic;

namespace MavericksBank.Models
{
    public partial class Transaction
    {
        public int TransactionId { get; set; }
        public int? SenderAccountId { get; set; }
        public int? RecieverAccountId { get; set; }
        public string? TransactionType { get; set; }
        public decimal? Amount { get; set; }
        public DateTime? TransactionDate { get; set; }

        public virtual Account? RecieverAccount { get; set; }
        public virtual Account? SenderAccount { get; set; }
    }
}

File: Mavericks Bank/server-side/src/app/account/transaction/transaction-history/transaction.ts
export class TransactionHistoryDetails {
    transactionId: number;
    senderAccountId: number | null;
    recieverAccountId: number | null;
    transactionType: string | null;
    amount: number;
    transactionDate: Date;
    recieverAccount: any | null;
    senderAccount: any | null;
  
    constructor(
      transactionId: number,
      senderAccountId: number | null,
      recieverAccountId: number | null,
      transactionType: string | null,
      amount: number,
      transactionDate: Date,
      recieverAccount: any | null,
      senderAccount: any | null
    ) {
      this.transactionId = transactionId;
      this.senderAccountId = senderAccountId;
      this.recieverAccountId = recieverAccountId;
      this.transactionType = transactionType;
      this.amount = amount;
      this.transactionDate = transactionDate;
      this.recieverAccount = recieverAccount;
      this.senderAccount = senderAccount;
    }
    refreshPage() {
        window.location.reload();
    }
    
  }
  